src/Components/Aspire.Azure.Messaging.WebPubSub/AspireWebPubSubExtensions.cs

```diff
@@ -0,0 +1,138 @@
+    /// Access the <see cref="WebPubSubServiceClient"/> using <see cref="FromKeyedServicesAttribute"/> with hubName as the key.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="connectionName">A name used to retrieve the connection string from the ConnectionStrings configuration section.</param>
+    /// <param name="hubName">The hub used for Web PubSub service, hubName is also considered as the name of the registered <see cref="WebPubSubServiceClient"/> as well as the serviceKey of the service</param>
+    /// <param name="configureSettings">An optional method that can be used for customizing the <see cref="AzureMessagingWebPubSubSettings"/>. It's invoked after the settings are read from the configuration.</param>
+    /// <param name="configureClientBuilder">An optional method that can be used for customizing the <see cref="IAzureClientBuilder{WebPubSubServiceClient, WebPubSubServiceClientOptions}"/>.</param>
+    /// <remarks>Reads the configuration from "Aspire.Azure.Messaging.WebPubSub" section.</remarks>
+    /// <exception cref="InvalidOperationException">Thrown when neither <see cref="AzureMessagingWebPubSubSettings.ConnectionString"/> nor <see cref="AzureMessagingWebPubSubSettings.Endpoint"/> is provided.</exception>
+    /// <example>
+    /// Use the following code to add a Web PubSub service client to the services collection:
+    /// <code>
+    /// builder.AddAzureWebPubSubHub("wps1", "chatHub");
+    /// </code>
+    /// </example>
+    /// <example>
+    /// Use the following code to use the registered service client in a controller:
+    /// <code>
+    /// public class HomeController : Controller
+    /// {
+    ///     private readonly WebPubSubServiceClient _service;
+    ///     public HomeController([FromKeyedServices("chatHub")]WebPubSubServiceClient service)
+    ///     {
+    ///         _service = service;
+    ///     }
+    /// }
+    /// </code>
+    /// </example>
+    public static void AddAzureWebPubSubHub(
+        this IHostApplicationBuilder builder,
+        string connectionName,
+        string hubName,
+        Action<AzureMessagingWebPubSubSettings>? configureSettings = null,
+        Action<IAzureClientBuilder<WebPubSubServiceClient, WebPubSubServiceClientOptions>>? configureClientBuilder = null)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(hubName);
+
+        // 1. ConnectionString reads from ConnectionStrings:{connectionName}
+        // 2. Configuration reads from Aspire:Azure:Messaging:WebPubSub:{connectionName}:{hubName}
+        string configurationSectionName = WebPubSubComponent.GetKeyedConfigurationSectionName($"{connectionName}:{hubName}", DefaultConfigSectionName);
+        new WebPubSubComponent(hubName).AddClient(builder, configurationSectionName, configureSettings, configureClientBuilder, connectionName, serviceKey: hubName);
```

Comment:
I thought we agreed to follow the common pattern of AddAzureWebPubSubServiceClient and AddKeyedAzureWebPubSubServiceClient. AddAzureWebPubSubServiceClient shouldn't register a "keyed" DI service, but instead just a normal DI service. And if you need to use more than 1, you can use AddKeyedAzureWebPubSubServiceClient.

Reply:
Hi @eerhardt  I remembered you helped adding some telemetry related changes to the PR right? Sadly after force push the changes were lost (forgot that I was using another machine to resolve conflict and forgot to sync the latest change :() Could you help with the change?